### ISSUE TITLE
ImportError: Failed to initialize: Bad git executable

### ISSUE THREAD

**Issue by joelfreybott** [beeware/beeware#32]
After updating GitPython on v0.3.0 the following errors came:
`(beeware-venv) joel@joels-pc:~/beeware-tutorial$ briefcase new
Traceback (most recent call last):
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/git/__init__.py", line 83, in <module>
    refresh()
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/git/__init__.py", line 73, in refresh
    if not Git.refresh(path=path):
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/git/cmd.py", line 278, in refresh
    raise ImportError(err)
ImportError: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/joel/beeware-tutorial/beeware-venv/bin/briefcase", line 7, in <module>
    from briefcase.__main__ import main
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/briefcase/__main__.py", line 3, in <module>
    from .cmdline import parse_cmdline
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/briefcase/cmdline.py", line 6, in <module>
    from briefcase.commands import DevCommand, NewCommand
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/briefcase/commands/__init__.py", line 1, in <module>
    from .build import BuildCommand  # noqa
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/briefcase/commands/build.py", line 5, in <module>
    from .base import BaseCommand, full_kwargs
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/briefcase/commands/base.py", line 17, in <module>
    from git import exc as git_exceptions
  File "/home/joel/beeware-tutorial/beeware-venv/lib/python3.6/site-packages/git/__init__.py", line 85, in <module>
    raise ImportError('Failed to initialize: {0}'.format(exc))
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh()

All git commands will error until this is rectified.

This initial warning can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|none|n|0: for no warning or exception
    - warn|w|warning|1: for a printed warning
    - error|e|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

`

**Comment by freakboy3742** [comment]
Thanks for the report! This has been reported previously, but I wrote it off thinking it had been fixed it - clearly it hasn't been.

The error message isn't very clear, but it does point at the root cause - Briefcase can't start `git`. The fix is to (install git)[https://git-scm.com/]. This isn't something that Briefcase can automate, because it isn't a Python-specific tool.

There's two underlying things we need to fix:
1) We need to explicitly list `git` as a dependency in the BeeWare tutorial.
2) We need to improve the error message in briefcase when git isn't available. I've opened beeware/briefcase#303 to track this improvement.
